node.go

```diff
@@ -602,3 +602,159 @@ func sortFields(fields []Field) {
 func compareFields(a, b Field) int {
 	return strings.Compare(a.Name(), b.Name())
 }
+
+func RewriteSchema(schema *Schema, replace func(path []string, node Node) Node) *Schema {
+	return NewSchema(schema.name, rewriteSchema(nil, schema.root, replace))
+}
+
+func rewriteSchema(path []string, node Node, replace func(path []string, node Node) Node) Node {
+	if node.Leaf() {
+		return replace(path, node)
+	}
+
+	// Types requiring special handling.
+	switch n := node.(type) {
+	case *structNode:
+		// Maintain struct information including field order and index.
+		fields := make([]structField, 0, len(n.fields))
+		for _, field := range n.fields {
+			node := rewriteSchema(append(path, field.name), field.Node, replace)
+			if node == nil {
+				continue
+			}
+			fields = append(fields, structField{
+				Node:  node,
+				name:  field.name,
+				index: field.index,
+			})
+		}
+		return &structNode{
+			gotype: n.gotype,
+			fields: fields,
+		}
+	case *groupField:
```

Comment:
groupField - We want to unwrap it and access the inner Node, but the Field interface doesn't have a method for this.  So we catch it here.  Without this, a groupField gets wrapped in another layer in the loop below and introduces errors.